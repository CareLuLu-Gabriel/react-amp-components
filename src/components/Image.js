import React, { Component } from 'react'
import { string, bool } from 'prop-types'

import { pick } from '../utils/misc'
import commonPropTypes from './commonPropTypes'

class Image extends Component {

  static propTypes = {
    ...commonPropTypes,
    src: string.isRequired,
    srcset: string,
    noscript: bool,
    alt: string
  }

  get fallback() {
    const { fallback } = this.props
    if (fallback) {
      return <div fallback="">{fallback}</div>
    } else {
      return null
    }
  }

  get noscript() {
    const { noscript, src, width, height, alt } = this.props
    if (noscript) {
      return (
        <noscript>
          <img {...{ src, width, height, alt } } />
        </noscript>
      )
    } else {
      return null
    }
  }

  get properties() {
    return pick(this.props, ...[
      ...Object.keys(commonPropTypes),
      'src',
      'srcset',
      'alt'
    ])
  }

  render() {

    return (
      <amp-img {...{ ...this.properties } }>
        {this.fallback}
        {this.noscript}
      </amp-img>
    )
  }
}

export default Image
