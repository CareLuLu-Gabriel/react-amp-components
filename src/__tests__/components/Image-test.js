import React from 'react'
import Image from '../../components/Image'
import { render } from '../test-utils'

let image

describe('Image', () => {
  beforeEach(() => {
    image = render(
      <Image src="https://placekitten.com/200/300" />
    )
  })

  it('matches the snapshot', () => {
    expect(image).toMatchSnapshot()
  })
})

