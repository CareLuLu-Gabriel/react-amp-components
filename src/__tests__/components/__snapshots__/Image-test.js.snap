// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches the snapshot 1`] = `"<amp-img src=\\"https://placekitten.com/200/300\\"></amp-img>"`;
